pkg/controllers/repositoryserver/server.go

```diff
@@ -0,0 +1,236 @@
+	if err != nil {
+		errors.Wrap(err, "Failed to list users from the Kopia repository")
+	}
+
+	userProfiles := []maintenance.KopiaUserProfile{}
+
+	err = json.Unmarshal([]byte(stdout), &userProfiles)
+	if err != nil {
+		errors.Wrap(err, "Failed to unmarshal user list")
+	}
+
+	// Get list of usernames from ServerListUserCommand output to update the existing data with updated password
+	existingUserHostList := sets.String{}
+	for _, userProfile := range userProfiles {
+		existingUserHostList.Insert(userProfile.Username)
+	}
+
+	userAccess := h.RepositoryServerSecrets.serverUserAccess.Data
+	serverAccessUsername := h.RepositoryServer.Spec.Server.UserAccess.Username
+	for hostname, password := range userAccess {
+		serverUsername := fmt.Sprintf(repoServerUsernameFormat, serverAccessUsername, hostname)
+		if existingUserHostList.Has(serverUsername) {
+			h.Logger.Info("User already exists, updating passphrase", "username", serverUsername)
+			// Update password for the existing user
+			cmd := command.ServerSetUser(
+				command.ServerSetUserCommandArgs{
+					CommandArgs: &command.CommandArgs{
+						RepoPassword:   repoPassword,
+						ConfigFilePath: command.DefaultConfigFilePath,
+						LogDirectory:   command.DefaultLogDirectory,
+					},
+					NewUsername:  serverUsername,
+					UserPassword: string(password),
+				})
+			stdout, stderr, err := kube.Exec(h.KubeCli, h.RepositoryServer.Namespace, h.RepositoryServer.Status.ServerInfo.PodName, repoServerPodContainerName, cmd, nil)
+			format.Log(h.RepositoryServer.Status.ServerInfo.PodName, repoServerPodContainerName, stdout)
+			format.Log(h.RepositoryServer.Status.ServerInfo.PodName, repoServerPodContainerName, stderr)
+			if err != nil {
+				errors.Wrap(err, "Failed to update existing user passphrase from the Kopia API server")
+			}
+			continue
+		}
+		cmd := command.ServerAddUser(
+			command.ServerAddUserCommandArgs{
+				CommandArgs: &command.CommandArgs{
+					RepoPassword:   repoPassword,
+					ConfigFilePath: command.DefaultConfigFilePath,
+					LogDirectory:   command.DefaultLogDirectory,
+				},
+				NewUsername:  serverUsername,
+				UserPassword: string(password),
+			})
+		stdout, stderr, err := kube.Exec(h.KubeCli, h.RepositoryServer.Namespace, h.RepositoryServer.Status.ServerInfo.PodName, repoServerPodContainerName, cmd, nil)
+		format.Log(h.RepositoryServer.Status.ServerInfo.PodName, repoServerPodContainerName, stdout)
+		format.Log(h.RepositoryServer.Status.ServerInfo.PodName, repoServerPodContainerName, stderr)
+		if err != nil {
+			return errors.Wrap(err, "Failed to add new user to the Kopia API server")
+		}
+	}
+
+	repoServerAddress, serverAdminUserName, serverAdminPassword, err := h.getServerDetails(ctx)
+	if err != nil {
+		return err
+	}
+	err = h.refreshServer(ctx, repoServerAddress, serverAdminUserName, serverAdminPassword)
+	if err != nil {
+		return errors.Wrap(err, "Failed to refresh repository server")
```

Comment:
[nit] Also noticed the use `repository server` and `Kopia API Server` interchangeably. Let's make it consistent